### PR TITLE
[oidc] Restrict actions to team owners WEB-284

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Ensures that only team owners can perform actions on OIDC Client Configs

## Related Issue(s)

Fixes #

## How to test

Unit tests

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-oidc-au11b8dbf2e7</li>
	<li><b>🔗 URL</b> - <a href="https://mp-oidc-au11b8dbf2e7.preview.gitpod-dev.com/workspaces" target="_blank">mp-oidc-au11b8dbf2e7.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/gitpod-db/go/team_membership.go [hunk 1]
```diff
@@ -40,18 +41,52 @@ const (
 	TeamMembershipRole_Member = TeamMembershipRole("member")
 )
 
-func ListTeamMembershipsForUserIDs(ctx context.Context, conn *gorm.DB, userIDs []uuid.UUID) ([]TeamMembership, error) {
```
Comment (easyCZ) [inline components/gitpod-db/go/team_membership.go:43]: This was unused, removed

File: components/gitpod-db/go/team_membership_test.go [hunk 2]
```diff
@@ -7,95 +7,55 @@ package db_test
...
+	"testing"
+
 	db "github.com/gitpod-io/gitpod/components/gitpod-db/go"
 	"github.com/gitpod-io/gitpod/components/gitpod-db/go/dbtest"
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
-	"testing"
 )
 
-func TestTeamMembership_WriteRead(t *testing.T) {
```
Comment (easyCZ) [inline components/gitpod-db/go/team_membership_test.go:17]: Likewise, test removed as method was unused

File: components/public-api-server/pkg/apiv1/oidc.go [hunk 3]
```diff
@@ -54,6 +54,20 @@ func (s *OIDCService) CreateClientConfig(ctx context.Context, req *connect.Reque
...
+	}
+
+	_, userID, err := s.getUser(ctx, conn)
+	if err != nil {
+		return nil, err
+	}
+
+	if authorizationErr := s.userIsOrgOwner(ctx, userID, organizationID); authorizationErr != nil {
+		return nil, authorizationErr
+	}
```
Comment (easyCZ) [inline components/public-api-server/pkg/apiv1/oidc.go:69]: After this lands, we can do a pass to merge these blocks together as we nearly always do all of them in this exact order, but I didn't want to pollute this PR with the refactor

File: components/public-api-server/pkg/apiv1/oidc_test.go [hunk 4]
```diff
@@ -43,19 +43,16 @@ var (
 		},
 	}
 
-	user = newUser(&protocol.User{})
+	user           = newUser(&protocol.User{})
+	organizationID = uuid.New()
```
Comment (easyCZ) [inline components/public-api-server/pkg/apiv1/oidc_test.go:47]: We re-use these two across the tests for the default scenario - with user being org owner. There are are extra tests which test the negative case

**Review by AlexTugarev — APPROVED** [review]
LGTM!
